backend/api/Services/MissionRunService.cs

```diff
@@ -114,9 +114,18 @@ public async Task<IList<MissionRun>> ReadMissionRunQueue(string robotId)
 
         public async Task<MissionRun?> ReadNextScheduledMissionRun(string robotId)
         {
-            return await GetMissionRunsWithSubModels()
+            var nextScheduledMissionRun = await GetMissionRunsWithSubModels()
+                .Where(missionRun => missionRun.Robot.Id == robotId && missionRun.Status == MissionStatus.Pending && missionRun.MissionRunPriority == MissionRunPriority.Localization)
                 .OrderBy(missionRun => missionRun.DesiredStartTime)
-                .FirstOrDefaultAsync(missionRun => missionRun.Robot.Id == robotId && missionRun.Status == MissionStatus.Pending);
+                .FirstOrDefaultAsync();
+
+
+            nextScheduledMissionRun ??= await GetMissionRunsWithSubModels()
+                    .Where(missionRun => missionRun.Robot.Id == robotId && missionRun.Status == MissionStatus.Pending)
+                    .OrderBy(missionRun => missionRun.DesiredStartTime)
+                    .FirstOrDefaultAsync();
+
+            return nextScheduledMissionRun;
```

Comment:
I like the code and the approach, but could it maybe be a bit more readable to have line 117 define a variable called something like "nextScheduledLocalisation" and then have an if statement instead of the ?? operator check if it is set? That way it would be clear that we first find the localisation mission, and if this does not exist we find the next mission run and return that.

Reply:
Thanks- Discussed with @oysand and we will move the functionality one level up in the event handler instead. Will require more refactoring.

Reply:
Will close this PR and make a new one with new approach.